server-sdk/src/main/java/com/amazonaws/ml/mms/servingsdk/http/Request.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.amazonaws.ml.mms.servingsdk.http;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This defines the request object given to the custom endpoint
+ */
+public interface Request {
+    /**
+     * Get all header names in the request object
+     * @return
+     */
+    List<String> getHeaderNames();
+
+    /**
+     * Get the URI of the request
+     * @return URI
+     */
+    String getRequestURI();
+
+    /**
+     * Get the path of the endpoint
+     * @return
+     */
+    String getRequestPath();
```

Comment:
Not sure if we need this

Reply:
True. We can remove this as well. The path is also defined by customers. I can remove this as well.

Reply:
I am removing this for now. I think this is similar to the method() call. We can add it when we need it.